ddb-client/src/main/java/org/opensearch/remote/metadata/client/impl/DDBOpenSearchClient.java

```diff
@@ -164,7 +164,7 @@ public CompletionStage<PutDataObjectResponse> putDataObjectAsync(
         Executor executor,
         Boolean isMultiTenancyEnabled
     ) {
-        final String id = request.id() != null ? request.id() : UUID.randomUUID().toString();
+        final String id = Strings.isNullOrEmpty(request.id()) ? UUID.randomUUID().toString() : request.id();
```

Comment:
what if request.id is like : `" "`. May be this is not common to happen. But to make sure we can do:

`final String id = Strings.isNullOrEmpty(request.id().trim()) ? UUID.randomUUID().toString() : request.id();`

we can use Apache Commons Lang's StringUtils.isBlank() too if available?

Reply:
Or we can use `trim()`

Reply:
Let's confirm what OpenSearch actually does in this case.

@MohammedAghil since you were originally wanting to do this and it's now a bit more complex can you take this over?  I'll close this PR and you can start a new one.